### PR TITLE
Add new validator class - alpha numeric spaces

### PR DESCRIPTION
**Pull request by flamisz**
### Description:

This new class will check if the validated field only contains alphanumeric characters and spaces. No special characters allowed.

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
I think we have unit tests for existing validators so we'd want to add one for this.

File: tests/PHPUnit/Unit/Validator/AlphaNumSpacesTest.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * @group AlphaNumSpacesTest
+ */
+class AlphaNumSpacesTest extends TestCase
+{
+    public function test_validate_success_strings()
+    {
+        $this->validate('onlyalpha');
+        $this->validate(123);
+        $this->validate('abc 123');
+        $this->validate('abc 123 and 0');
```
Comment (sgiehl) [inline tests/PHPUnit/Unit/Validator/AlphaNumSpacesTest.php]: might be good to also test edge cases like German umlauts, some Chinese characters and maybe even some utf8 emojis maybe

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: tests/PHPUnit/Unit/Validator/AlphaNumSpacesTest.php (hunk 1 above)
Comment (flamisz) [reply tests/PHPUnit/Unit/Validator/AlphaNumSpacesTest.php]: definitely, I will do that 👍

Comment (flamisz) [reply tests/PHPUnit/Unit/Validator/AlphaNumSpacesTest.php]: I added more test cases with special characters. Emoji is not supported with this regex. Do you think we should let emojis as well?

File: tests/PHPUnit/Unit/Validator/AlphaNumSpacesTest.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+class AlphaNumSpacesTest extends TestCase
+{
+    public function test_validate_success_english_strings()
+    {
+        $this->validate('onlyalpha');
+        $this->validate(123);
+        $this->validate('abc 123');
+        $this->validate('ABC 123 and 0');
+
+        $this->assertTrue(true);
```
Comment (sgiehl) [inline tests/PHPUnit/Unit/Validator/AlphaNumSpacesTest.php:28]: might be better to use `$this->expectNotToPerformAssertions();` instead of doing an assert that can't fail.
Also you could also use one test for all strings with a dataprovider, like for the failing ones.

**Comment by flamisz** [comment]
I close this now as we don't need it.
